src/main/java/fi/aalto/cs/apluscourses/intellij/utils/CourseFileManager.java

```diff
@@ -216,5 +304,29 @@ private void loadFromJsonObject(@NotNull JSONObject jsonObject) throws IOExcepti
       this.modulesMetadata
           .put(moduleName, new ModuleMetadata(moduleVersion, downloadedAt));
     }
+
+
+    JSONObject dependenciesObject = jsonObject.optJSONObject(MODULE_DEPENDENCIES_KEY);
+    if (dependenciesObject != null) {
+      Iterable<String> depModuleNames = dependenciesObject::keys;
+
+      for (String moduleName : depModuleNames) {
+        Set<String> dependencies = new HashSet<>();
+        JSONArray jsonDependencies = dependenciesObject.getJSONArray(moduleName);
+        for (int i = 0; i < jsonDependencies.length(); i++) {
+          dependencies.add(jsonDependencies.getString(i));
+        }
+
+        moduleDependencies.put(moduleName, dependencies);
+      }
+    }
+    addAllDependencies();
+    fixDependencies();
+
+    writeCourseFile(new JSONObject()
+        .put(URL_KEY, courseUrl.toString())
+        .put(LANGUAGE_KEY, language)
+        .put(MODULES_KEY, modulesObject)
+        .put(MODULE_DEPENDENCIES_KEY, createDependenciesObject()));
   }
```

Comment:
Why is the course file written here? Anyways, this duplicates the code in lines 135-139.

Reply:
So that users with existing modules have their dependencies saved without needing to download another module